ge/org.osate.ge.ba/src/org/osate/ge/ba/ui/properties/EmbeddedTextEditor.java

```diff
@@ -24,41 +24,48 @@
 package org.osate.ge.ba.ui.properties;
 
 import org.eclipse.core.resources.IProject;
-import org.eclipse.jface.layout.GridDataFactory;
 import org.eclipse.jface.layout.GridLayoutFactory;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.custom.StyledText;
 import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.layout.GridData;
 import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Caret;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Control;
 import org.osate.ge.swt.SwtUtil;
 
-public class EmbeddedTextControls extends Composite {
-	private final StyledText styledText;
-	private final Button btn;
+public class EmbeddedTextEditor extends Composite {
+	private StyledText styledText;
+	private Button btn;
 	private EmbeddedXtextAdapter xtextAdapter;
+	private final int styledTextStyle;
+	private final GridData styledTextLayoutData;
 
-	public EmbeddedTextControls(final Composite parent, final int style, final int styledTextStyle, final int btnStyle,
-			final String editBtnText) {
+	public EmbeddedTextEditor(final Composite parent, final int style, final int styledTextStyle,
+			final GridData styledTextLayoutData) {
 		super(parent, style);
 		this.setBackground(parent.getBackground());
 		this.setLayout(GridLayoutFactory.swtDefaults().numColumns(2).create());
-		this.setLayoutData(GridDataFactory.swtDefaults().align(SWT.FILL, SWT.FILL).grab(true, true)
-				.hint(SWT.DEFAULT, SWT.DEFAULT).create());
+		this.styledTextStyle = styledTextStyle;
+		this.styledTextLayoutData = styledTextLayoutData;
+	}
 
+	public void refresh() {
```

Comment:
Consider renaming to createControls(). Document when this should be called. Consider allowing specifying the testing IDs as parameters. It is odd that must call set*TestId multiple times. This class is leaking what should be internal details into its API. Alternatively, the class could store the testing IDs and automatic apply and reapply them as needed. (Shouldn't matter when set*TestId is called but it should only need to be called once).

Reply:
Good point.  I fixed this.